sdkbuild/typescript.go

```diff
@@ -0,0 +1,226 @@
+package sdkbuild
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// BuildTypeScriptProgramOptions are options for BuildTypeScriptProgram.
+type BuildTypeScriptProgramOptions struct {
+	// Directory that will have a temporary directory created underneath.
+	BaseDir string
+	// Required version. If it contains a slash it is assumed to be a path with a
+	// package.json. Otherwise it is a specific version (with leading "v" is
+	// trimmed if present).
+	Version string
+	// Required set of paths to include in tsconfig.json paths for the harness.
+	// The paths should be relative to one-directory beneath BaseDir.
+	HarnessPaths map[string][]string
```

Comment:
Can we make this more generic and not require the concept of a harness?

Reply:
Yeah, just call this `AdditionalTsConfigPaths`.

Reply:
Harness is just an English term. I guess I can change it to `AdditionalTSConfigPaths`

Reply:
The `Additional` makes it seem optional, but `TSConfigPaths`, ok